components/apim_appgw/main.tf

```diff
@@ -45,6 +45,7 @@ module "app-gw" {
   exclusions                                   = var.apim_appgw_exclusions
   public_ip_enable_multiple_availability_zones = true
   project_name                                 = var.project
+  usage_name                                   = "${var.project}-apim"
```

Comment:
Dont think this is needed?

Reply:
usage_name field is being used to add user defined managed identity, if we do not supply name, it will throw error on pipline that the resouce has already been deployed

Reply:
sorry, missed that requirement. Was looking at another area in the module.